File: javascript/remove_element.js
//
// Created by yash on 09/05/21.
//
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    let len = nums.length;
    let i = 0;

    while (i < len) {
        if (nums[i] == val) {
            nums[i] = nums[len - 1];
            len--;
        } else {
            i++;
        }
    }
    return len;
};

let nums = [3,2,2,3]
let val = 3
removeElement(nums,val);
console.log(nums);

File: cpp/remove_element.cpp
//
// Created by yash on 09/05/21.
//

#include <iostream>
#include <vector>
using namespace std;


class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int length {(int)nums.size()};
        int i {};

        while (i < length) {
            if (nums[i] == val) {
                nums[i] = nums[length - 1];
                length--;
            } else {
                i++;
            }
        }

        return length;
    }
};

int main() {
    int n;
    cin >> n;
    vector<int> nums;
    for (int i{}; i<n; i++) {
        int x;
        cin >> x;
        nums.push_back(x);
    }

    int val;
    cin >> val;

    Solution s;
    int ans = s.removeElement(nums, val);
    cout << ans << endl;
}